torch/_inductor/ir.py

```diff
@@ -5442,9 +5442,22 @@ def _original_deconv_weight_size(
 
         req_stride_order = [0] + list(reversed(range(1, len(stride) + 1)))
         req_stride_order = [len(req_stride_order)] + req_stride_order
-        output_stride = make_channels_last_strides_for(output_size)
 
     x = cls.require_stride_order(x, req_stride_order)
+
+    # We won't do weight prepack for Conv if dynamic_shapes.
+    # In static shape cases, since weight is prepacked, we'll always force output to be channels last in the Conv kernel.
+    # In dynamic shape cases, for input with channels = 1, like tensor of size (s0, 1, 28, 28) and stride (784, 784, 28, 1),
+    # x = cls.require_stride_order(x, req_stride_order) where req_stride_order is in the channels last order
+    # won't change the stride of this tensor since stride for dimensions of size 1 is ignored. While in Conv kernel,
+    # this tensor is considered as channels first and the output will be in contiguous format.
+    # To align the behavior of the Conv kernel, we set the output_stride in such case to be contiguous instead of channels last.
+    dynamic_shapes = not all(isinstance(i, int) for i in (output_size))
+    if dynamic_shapes and is_contiguous_storage_and_layout(x):
```

Comment:
Instead of checking `dynamic_shapes` condition, is it more direct to check if the weight is packed or not here? The logic is that channels last format for output is applied when input is channels last or weight is packed. Whether we prepack weights on `dynamic_shapes` is separate.

Reply:
If weight is packed, the `weight` here is a `TensorBox` with `FixedLayout('cpu', torch.float32, size = [3, 1, 3, 3], stride = [1, 0, 0, 0])`.
If not prepacked, `weight` is `TensorBox` with `FixedLayout('cpu', torch.float32, size = [3, 1, 3, 3], stride = [9, 9, 3, 1])`.

I tried just now but seems there's no API like `weight.is_mkldnn()` here to decide whether it's packed or not and we don't have such information in the `FixedLayout` either. We may infer from the stride `[1, 0, 0, 0]` that it's likely to be packed but I feel like it might not be accurate to infer whether it's packed or not just from the stride value, since it's possible for a plain format tensor to have stride like `[1, 0, 0, 0]` as well.